### PR TITLE
Fix nil pointer dereference  error when open link with invalid pull index

### LINKED ISSUE #26331: PANIC: runtime error: invalid memory address or nil pointer dereference
### Description

Open link with invalid pull index, like
https://example.com/test/repo-name/pulls/99999

```log
An error occurred:

PANIC: runtime error: invalid memory address or nil pointer dereference
/usr/local/go/src/runtime/panic.go:884 (0x44a7f2)
/go/src/code.gitea.io/gitea/modules/web/routing/logger_manager.go:116 (0x1cd2264)
/usr/local/go/src/runtime/panic.go:884 (0x44a7f2)
/usr/local/go/src/runtime/panic.go:260 (0x4621fc)
/usr/local/go/src/runtime/signal_unix.go:841 (0x4621cc)
/go/src/code.gitea.io/gitea/routers/web/repo/pull.go:365 (0x26c5e33)
/usr/local/go/src/reflect/value.go:586 (0x4b7dca)
/usr/local/go/src/reflect/value.go:370 (0x4b707b)
/go/src/code.gitea.io/gitea/modules/web/handler.go:166 (0x1cdb765)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/modules/web/handler.go:176 (0x1cdb7fb)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/modules/web/handler.go:176 (0x1cdb7fb)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/modules/web/handler.go:176 (0x1cdb7fb)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/modules/web/handler.go:176 (0x1cdb7fb)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/modules/web/handler.go:176 (0x1cdb7fb)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.8/chain.go:31 (0x1cd2dcb)
/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.8/mux.go:444 (0x1cd5cd5)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/modules/web/handler.go:176 (0x1cdb7fb)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/modules/web/handler.go:176 (0x1cdb7fb)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/modules/web/handler.go:176 (0x1cdb7fb)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.8/middleware/get_head.go:37 (0x2595ead)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/modules/web/handler.go:145 (0x1cdba2b)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/modules/web/handler.go:176 (0x1cdb7fb)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/modules/context/context.go:204 (0x1dedd8b)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/modules/web/handler.go:145 (0x1cdba2b)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/pkg/mod/gitea.com/go-chi/session@v0.0.0-20230613035928-39541325faa3/session.go:257 (0x1d4078a)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/modules/web/handler.go:145 (0x1cdba2b)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.8/mux.go:73 (0x1cd3a54)
/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.8/mux.go:316 (0x1cd5463)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.8/mux.go:444 (0x1cd5cd5)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.8/mux.go:73 (0x1cd3a54)
/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.8/mux.go:316 (0x1cd5463)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.8/mux.go:444 (0x1cd5cd5)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/modules/web/routing/logger_manager.go:122 (0x1cd2093)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/modules/web/handler.go:145 (0x1cdba2b)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/pkg/mod/github.com/chi-middleware/proxy@v1.1.1/middleware.go:37 (0x25095d6)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/modules/web/handler.go:145 (0x1cdba2b)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/routers/common/middleware.go:45 (0x250c714)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/modules/web/handler.go:145 (0x1cdba2b)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/routers/common/middleware.go:37 (0x250c22f)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/modules/web/handler.go:145 (0x1cdba2b)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/routers/common/middleware.go:99 (0x250b7a9)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/src/code.gitea.io/gitea/modules/web/handler.go:145 (0x1cdba2b)
/usr/local/go/src/net/http/server.go:2122 (0x9cd7ce)
/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.8/mux.go:90 (0x1cd3a0f)
/go/src/code.gitea.io/gitea/modules/web/route.go:171 (0x1cdcd0d)
/usr/local/go/src/net/http/server.go:2936 (0x9d0dd5)
/usr/local/go/src/net/http/server.go:1995 (0x9cc2f1)
/usr/local/go/src/runtime/asm_amd64.s:1598 (0x482300)

```

### Gitea Version

latest dev

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

https://try.gitea.io/doktormike/Helloyou/pulls/9999

### Database

None

### PR DESCRIPTION
**Pull request by CaiCandong**
fix #26331 

Before:
image

After:
image

### CODE REVIEW

File: routers/web/repo/pull.go [hunk 1]
```diff
@@ -362,6 +362,9 @@ func setMergeTarget(ctx *context.Context, pull *issues_model.PullRequest) {
 // GetPullDiffStats get Pull Requests diff stats
 func GetPullDiffStats(ctx *context.Context) {
 	issue := checkPullInfo(ctx)
+	if issue == nil {
```
Comment (lunny) [inline routers/web/repo/pull.go]: Maybe move line 372->374 here?

Comment (CaiCandong) [reply routers/web/repo/pull.go]: I don't think the readability of using `ctx.Written()` is very good ，Perhaps we could add a `bool` return value to `checkPullInfo` to determine if the check passes?
```
func checkPullInfo(ctx *context.Context) (*issues_model.Issue, bool) {
```
```
	issue, ok := checkPullInfo(ctx)
	if !ok {
		return
	}
```

Comment (lunny) [reply routers/web/repo/pull.go]: > I don't think the readability of using `ctx.Written()` is very good ，Perhaps we could add a `bool` return value to `checkPullInfo` to determine if the check passes?
> 
> ```
> func checkPullInfo(ctx *context.Context) (*issues_model.Issue, bool) {
> ```
> 
> ```
> 	issue, ok := checkPullInfo(ctx)
> 	if !ok {
> 		return
> 	}
> ```

That's also OK.

File: routers/web/repo/pull.go [hunk 2]
```diff
@@ -299,51 +299,51 @@ func ForkPost(ctx *context.Context) {
 	ctx.Redirect(ctxUser.HomeLink() + "/" + url.PathEscape(repo.Name))
 }
 
-func checkPullInfo(ctx *context.Context) *issues_model.Issue {
+func getPullInfo(ctx *context.Context) (*issues_model.Issue, bool) {
```
Comment (delvh) [inline routers/web/repo/pull.go]: What about named return values like
```suggestion
func getPullInfo(ctx *context.Context) (issue *issues_model.Issue, ok bool) {
```
?

**Review by delvh — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**